Fixes #20264: Update plugin title rendering with default icon

pheus opened the pull request:
### Fixes: #20264

<!-- Please include a summary of the proposed changes below. -->

**Summary**
- Replace inline plugin title markup with a reusable template in `template_code.py`.
- Add a **default core SVG icon** used when no plugin icon is provided.
- Keep `title_long` as a `TemplateColumn` using the template; remove redundant `render_title_long()`.
- Correct `order_by` from `name` → `title_long`.

**Impact**
- UI-only; no API/DB changes.
- Deduplicates logic and ensures consistent icon rendering.
- Catalog/author-supplied `icon_url` still overrides the default.

bctiemann commented on netbox/core/tables/plugins.py:
```diff
@@ -93,10 +85,4 @@ class Meta(BaseTable.Meta):
-            return value
-        url = reverse('core:plugin', args=[record.config_name])
-        return mark_safe(f"<a href='{url}'>{value}</a>")
+        order_by = ('-is_installed', '-is_certified', 'title_long')
```
Actually I guess this depends on https://github.com/netbox-community/netbox/pull/20267, with which it would make sense, right?

pheus replied:
Thanks for the review!

You’re right: `order_by="-is_installed"` predates this PR and is currently broken because there’s no concrete field. That will be fixed by #20267.  
This PR only corrects the bad reference by changing `order_by="name"` to `order_by="title_long"`.

pheus commented:
Thanks for calling this out!

A few options to make this consistent:

1) **Fallback default (preferred):** ship a small default SVG “box” in NetBox core and render it when a plugin provides no icon. Consistent look, no new dependency, easy to size.
2) **Standardize on MDI:** use the Material Design Icon as the default for all plugins, letting plugins override with their own icon, but sizing questions remain.
3) **No icon:** hide the icon like it's currently implemented.

Happy to implement whichever you prefer.

bctiemann commented:
I like the idea of option 1 I think.

It's interesting that the current default "box+" icon suggests that you're supposed to click on it to install the plugin, but it's not clickable and does nothing; maybe the default "box" you're suggesting should take its place? I assume a catalog icon can override the icon even if it's not installed?

pheus committed:
feat(core): Update plugin title rendering with default icon

Replaces inline plugin title HTML with a reusable template in
`template_code.py`. Adds a default icon for plugins without custom icons
and updates the table logic to use this template.
Removes redundant logic from the `render_title_long` method to improve
maintainability.
Changes the `order_by` field in `plugins.py` from `name` to
`title_long`.

Fixes #20264
